File: spec/lua/test_switch_repeat_expr.lua
local luaunit = require("luaunit")

require("switch_repeat_expr")

TestSwitchRepeatExpr = {}

function TestSwitchRepeatExpr:test_switch_repeat_expr()
    local r = SwitchRepeatExpr:from_file("src/switch_tlv.bin")

    luaunit.assertEquals(r.code, 0x11)
    luaunit.assertEquals(r.size, 9)
    luaunit.assertEquals(r.body[1].first, "Stuff\000Me\000")
end

File: spec/javascript/test_switch_repeat_expr.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('SwitchRepeatExpr', 'src/switch_tlv.bin', function(r) {
  assert.equal(r.code, 0x11);
  assert.equal(r.size, 9);
  assert.equal(KaitaiStream.bytesToStr(r.body[0].first, "UTF-8"), "Stuff\0Me\0");
});
